### PR TITLE
:sparkles: a client builder for manifestworks

### PR DESCRIPTION
**Pull request by skeeey**
## Summary
a client builder for manifestworks

## Related issue(s)

Fixes #

### CODE REVIEW

File: cloudevents/work/clientbuilder.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+		return nil, err
+	}
+
+	manifestWorkClient := agentclient.NewManifestWorkAgentClient(cloudEventsClient, watcher)
+	workClient := &internal.WorkV1ClientWrapper{ManifestWorkClient: manifestWorkClient}
+	workClientSet := &internal.WorkClientSetWrapper{WorkV1ClientWrapper: workClient}
+	factory := workinformers.NewSharedInformerFactoryWithOptions(workClientSet, b.informerResyncTime, b.informerOptions...)
+	informers := factory.Work().V1().ManifestWorks()
+	manifestWorkLister := informers.Lister()
+	namespacedLister := manifestWorkLister.ManifestWorks(b.clusterName)
```
Comment (qiujian16) [inline cloudevents/work/clientbuilder.go:139]: I am worried that we have multiple thread write cache.Store at the same time. Is cache.Store thread safe? What happens if a thread is updating the spec and another is updating the status or finallizer?

Comment (skeeey) [reply cloudevents/work/clientbuilder.go:139]: the informer cache is a [thread safe store](https://github.com/kubernetes/client-go/blob/v0.26.1/tools/cache/store.go#L273), when we update the spec/status/finallizer, all of these actions will be handled by the [reflector](https://github.com/kubernetes/client-go/blob/v0.26.1/tools/cache/reflector.go#L530) , the reflector will call its store to [update](https://github.com/kubernetes/client-go/blob/v0.26.1/tools/cache/thread_safe_store.go#L221) them, so I think it should be thread safe call

Comment (qiujian16) [reply cloudevents/work/clientbuilder.go:139]: I think we should add some note here that client and informer is sharing the store in this implementation. We may need to revisit the implementation in the future. Ideally, the store should be only written from the server.

Comment (skeeey) [reply cloudevents/work/clientbuilder.go:139]: OK, added a todo here

**Comment by qiujian16** [comment]
/approve

leave to @morvencao to tag

**Review by morvencao — COMMENTED** [review]
💯 
/lgtm

**Pull request merged**